### PR TITLE
WIP: Add a build test for Android, iOS and macOS to our CI

### PR DESCRIPTION
**Pull request by nilsreichardt**
## Description
In the past, we had several times problems with building for specific platforms (see #161, #64). To prevent this on a PR level, we check for every pull request, if the build is still possible.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -55,8 +55,8 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-go@v2
         with:
-          go-version: '^1.13.1'
-      - run: go install github.com/google/addlicense@latest 
+          go-version: "^1.13.1"
```
Comment (nilsreichardt) [inline .github/workflows/main.yml:58]:
```suggestion
          go-version: '^1.13.1'
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -115,9 +115,69 @@ jobs:
       - name: Deploy to Firebase Hosting (sharezone-debug)
         uses: FirebaseExtended/action-hosting-deploy@v0
         with:
-          repoToken: '${{ secrets.GITHUB_TOKEN }}'
-          firebaseServiceAccount: '${{ secrets.FIREBASE_SERVICE_ACCOUNT_SHAREZONE_DEBUG }}'
+          repoToken: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (nilsreichardt) [inline .github/workflows/main.yml:118]:
```suggestion
          repoToken: '${{ secrets.GITHUB_TOKEN }}'
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -115,9 +115,69 @@ jobs:
...
+          firebaseServiceAccount: "${{ secrets.FIREBASE_SERVICE_ACCOUNT_SHAREZONE_DEBUG }}"
           projectId: sharezone-debug
           entryPoint: "./app"
-          expires: '7d'
-          target: 'test-web-app'
+          expires: "7d"
```
Comment (nilsreichardt) [inline .github/workflows/main.yml:122]:
```suggestion
          expires: '7d'
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -115,9 +115,69 @@ jobs:
...
+          target: "test-web-app"
```
Comment (nilsreichardt) [inline .github/workflows/main.yml:123]:
```suggestion
          target: 'test-web-app'
```

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -116,8 +116,68 @@ jobs:
         uses: FirebaseExtended/action-hosting-deploy@v0
         with:
           repoToken: '${{ secrets.GITHUB_TOKEN }}'
-          firebaseServiceAccount: '${{ secrets.FIREBASE_SERVICE_ACCOUNT_SHAREZONE_DEBUG }}'
+          firebaseServiceAccount: "${{ secrets.FIREBASE_SERVICE_ACCOUNT_SHAREZONE_DEBUG }}"
```
Comment (nilsreichardt) [inline .github/workflows/main.yml:119]:
```suggestion
          firebaseServiceAccount: '${{ secrets.FIREBASE_SERVICE_ACCOUNT_SHAREZONE_DEBUG }}'
```

**Comment by nilsreichardt** [comment]
Seems to need a bit more investigation :(

**Comment by Jonas-Sander** [comment]
Should probably also not run these on WIP PRs?

**Comment by Jonas-Sander** [comment]
Btw, haven't looked at the errors but there is also the `act` CLI to run actions/jobs locally. This might help you to debug at least some issues.

**Comment by nilsreichardt** [comment]
> Should probably also not run these on WIP PRs?

Yeah, I can add this 👍

**Comment by nilsreichardt** [comment]
I will close this PR because we have iOS integration tests and will Android integration tests (https://github.com/SharezoneApp/sharezone-app/pull/230) which are building the app. Building the app for macOS is a bit more complex because it requires code signing.

**Pull request closed without merging**